sdk/reservations/Azure.ResourceManager.Reservations/tests/Tests/GetCatalogTests.cs

```diff
@@ -0,0 +1,495 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Reservations.Models;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Reservations.Tests
+{
+    public class GetCatalogTests : ReservationsManagementClientBase
+    {
+        public GetCatalogTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [SetUp]
+        public async Task ClearAndInitialize()
+        {
+            if (Mode == RecordedTestMode.Record || Mode == RecordedTestMode.Playback)
+            {
+                await InitializeClients();
+            }
+        }
+
+        [TestCase]
+        [RecordedTest]
+        public async Task TestGetCatalogForVirtualMachines()
+        {
+            AsyncPageable<ReservationCatalog> catalogResponse = Subscription.GetCatalogAsync("VirtualMachines", "eastus");
+            List<ReservationCatalog> catalogResult = await catalogResponse.ToEnumerableAsync();
+
+            Assert.NotNull(catalogResult);
+            Assert.IsTrue(catalogResult.Count > 0);
+
+            catalogResult.ForEach(item =>
+            {
+                Assert.AreEqual("virtualMachines", item.ReservedResourceType);
+                Assert.NotNull(item.Locations);
+                Assert.IsTrue(item.Locations.Count == 1);
+                Assert.AreEqual("eastus", item.Locations[0].Name);
+                Assert.AreEqual("East US", item.Locations[0].DisplayName);
+            });
```

Comment:
Can we extract this code to a private method? I found they are duplicated in many tests.

Reply:
will do